### PR TITLE
2017Q1.feature.socketcan

### PR DESCRIPTION
**Pull request by chrberger**
* Added SocketCAN implementation
* TODO: Test & refactor CANDevice to CharCANDevice implementation

### CODE REVIEW

File: automotive/odcantools/libodcantools/src/SocketCANDevice.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            // Close file descriptor.
+            if (m_socketCAN > -1) {
+                close(m_socketCAN);
+            }
+
+            CLOG << "done." << endl;
+        }
+
+        bool SocketCANDevice::isOpen() const {
+            return (m_socketCAN != NULL);
```
Comment (chrberger) [inline automotive/odcantools/libodcantools/src/SocketCANDevice.cpp]: Check for >-1

File: automotive/odcantools/libodcantools/src/SocketCANDevice.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+        void SocketCANDevice::run() {
+#ifdef __linux__
+            struct can_frame frame;
+            fd_set rfds;
+            struct timeval timeout;
+            struct timeval socketTimeStamp;
+            int32_t nbytes = 0;
+
+            serviceReady();
```
Comment (chrberger) [inline automotive/odcantools/libodcantools/src/SocketCANDevice.cpp]: serviceReady() must be outside ifdef

File: automotive/odcantools/libodcantools/src/SocketCANDevice.cpp (hunk 1 above)
Comment (chrberger) [reply automotive/odcantools/libodcantools/src/SocketCANDevice.cpp]: Fixed

File: automotive/odcantools/libodcantools/src/SocketCANDevice.cpp (hunk 2 above)
Comment (chrberger) [reply automotive/odcantools/libodcantools/src/SocketCANDevice.cpp]: Fixed

**Review by fgiaimo — APPROVED** [review]

**Pull request merged**